### ISSUE TITLE
`-reloaddirs` option does not work

### ISSUE THREAD

**Issue by smac89** [wailsapp/wails#2829]
### Description

When using the `-reloaddirs` option to specify a folder outside the working directory, the app is not reloaded when any files in those folders change

### To Reproduce

1. `mkdir -p /tmp/myproject/common`
2. `cd /tmp/myproject && go work init`
4. `wails init -n gui -t preact-ts`
5. `go work use gui`
7. `cd common && go mod init myproject/common`
8. `go work use .`
9. `touch foo.go`
10. copy the following to `foo.go`
    ```go
    package common
    
    import "fmt"
   
    func doSomething() {
	    fmt.Println("Something else")
    }
    ```
11. `cd /tmp/myproject/gui`
12. `wails dev -reloaddirs=/tmp/myproject/common`
13. In a different process, make changes to `/tmp/myproject/common/foo.go`

### Expected behaviour

Changes made to `/tmp/myproject/common/foo.go` should have caused the watcher to reload, but instead nothing happens

### Attempted Fixes

I've tried specifying the `reloaddirs` option in many ways, but none of them works. I believe the problem is that the variable which contains `dirsThatTriggerAReload`, found here: https://github.com/wailsapp/wails/blob/7fae22b50ceaa4750663f2289957dee0a1f5bdb9/v2/cmd/wails/internal/dev/dev.go#L339-L343

is not being added to the watcher, therefore none of the changes specified by `-reloaddirs` option are being detected

### System Details

```shell
Wails CLI v2.5.1

# System

OS           | ArcoLinux
Version      | Unknown  
ID           | arcolinux
Go Version   | go1.20.5 
Platform     | linux    
Architecture | amd64    

# Wails

Version         | v2.5.1
Package Manager | pacman

# Dependencies

Dependency | Package Name | Status    | Version    
*docker    | docker       | Installed | 1:24.0.5-1 
gcc        | gcc          | Installed | 13.2.1-3   
libgtk-3   | gtk3         | Installed | 1:3.24.38-1
libwebkit  | webkit2gtk   | Installed | 2.40.5-1   
npm        | npm          | Installed | 9.8.1-1    
pkg-config | pkgconf      | Installed | 1.8.1-1    
* - Optional Dependency

# Diagnosis

Your system is ready for Wails development!
 ♥   If Wails is useful to you or your company, please consider sponsoring the project:
https://github.com/sponsors/leaanthony
```

**Comment by smac89** [comment]
@haukened I don't think so. The directories I'm having problems with are those that exist outside the project directory (but within the go workspace).

As I observed in my attempts to fix the issue, I believe the problem is that extra directories specified with the `-reloaddirs` option are not being added to the watch list. I haven't had time to fully test this, but that's my theory for now.

**Comment by haukened** [comment]
Ok, looking at this i can confirm that its a bug. It looks like the `*fsnotify.Watcher` is never informed of these directories.  These exists logic to cause a reload:
```
for _, reloadDir := range dirsThatTriggerAReload {
    if strings.HasPrefix(itemName, reloadDir) {
        reload = true
        break
    }
}
```
but without notifying the watcher that this directory exists, [the required `watcher.Event` ](https://github.com/wailsapp/wails/blob/4ad4d7c7e3dd9977c6f931b3bc547dcf94456412/v2/cmd/wails/internal/dev/dev.go#L378)is never generated.

Currently, they are even passed to initialize watcher: [`func initialiseWatcher(cwd string) (*fsnotify.Watcher, error) {}`](https://github.com/wailsapp/wails/blob/4ad4d7c7e3dd9977c6f931b3bc547dcf94456412/v2/cmd/wails/internal/dev/watcher.go#L20).

**Comment by haukened** [comment]
Fixed by #2871

**Comment by smac89** [comment]
That's awesome! Thanks for your time investigating this.
